src/audio/external.c

```diff
@@ -2143,29 +2078,21 @@ void set_sound_moving_speed(u8 bank, u8 speed) {
 /**
  * Called from threads: thread5_game_loop
  */
-void play_dialog_sound(u8 dialogID) {
-    u8 speaker;
-
-    if (dialogID >= DIALOG_COUNT) {
-        dialogID = 0;
-    }
-
-    speaker = sDialogSpeaker[dialogID];
-    if (speaker != 0xff) {
-        play_sound(sDialogSpeakerVoice[speaker], gGlobalSoundSource);
-
-        // Play music during bowser message that appears when first entering the
-        // castle or when trying to enter a door without enough stars
-        if (speaker == BOWS1) {
-            seq_player_play_sequence(SEQ_PLAYER_ENV, SEQ_EVENT_KOOPA_MESSAGE, 0);
+void play_dialog_sound(s32 voice) {
+    if (voice != NO_SOUND && voice != 1) {
+        // "You've stepped on the (Wing|Metal|Vanish) Cap Switch"
+        if (voice == SEQ_EVENT_SOLVE_PUZZLE) {
```

Comment:
ok so what do you want me to do here

Reply:
I guess Toad jingle isn't strictly an improvement since those are behavior based anyway. Personally I would suggest prioritizing the removal of the hardcoded dialog sequences and then add something for music afterwards only if it still makes sense.

The other option would be to play the puzzle jingle as part of the cap switch behavior instead so we don't have a weird edge case like this in this function (this might be a better option to be honest).